src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,63 @@
+package tasks;
+
+/**
+ * Tasks.Task represents a generic task that subclasses can inherit from. Represents
+ * something that needs to be done.
+ * <p>
+ * This class should not be used on its own, but inherited by other subclasses and
+ * used there.
+ */
+public abstract class Task {
+    private final String taskDescription;
+    private boolean isComplete = false; // Initialized to false by default
+
+    public Task(String taskDescription) {
+        this.taskDescription = taskDescription;
+    }
+
+    /**
+     * Abstract function to be overridden representing a Task's internal representation
+     * in storage.
+     *
+     * @return String representing internal representation of a Task.
+     */
+    public abstract String getEncodedValue();
+
+    /**
+     * Returns a Task's description.
+     *
+     * @return String representing a Tasks's description.
+     */
+    public String getTaskDescription() {
+        return this.taskDescription;
+    }
+
+    /**
+     * Returns whether a Task is completed.
+     *
+     * @return Boolean representing if a Task is complete.
+     */
+    public boolean getIsComplete() {
```

Comment:
Personally, I am not sure if the method name is acceptable. I get that you are using a getter method, but for a boolean, shouldn't the name just be IsComplete()?